### PR TITLE
Support for Visual Studio 2015

### PR DESCRIPTION
**Pull request by jimmason**
Apologies for the out of the blue PR. Just a small change that adds Visual Studio 2015 as a supported product to the vsixmanifest so that the templates can be installed in the latest version.

### CODE REVIEW

**Comment by danyx23** [comment]
:+1: 
Can we get that merged? Would be useful to have up to date templates in vs 2015 working...

**Comment by devlead** [comment]
Indeed:+1:, perhaps rebase into one commit would make it nicer for the maintainers to review and merge.

**Comment by jimmason** [comment]
Yes, sorry its a bit of a mess, i will get this rebased into a single commit. Ended up commiting a load of stuff that needed to be in another branch and had to revert.

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,4 @@ _[Rr]e[Ss]harper.*/
 src/_NCrunch_Nancy/
 Thumbs.db
 .idea
+/src/.vs
```
Comment (devlead) [inline .gitignore]: I'm not the maintainer of this repo, but I am of other projects, in general I would try to change as little as possible and keep commits cleans, in general I wouldn't touch repo/project settings when adding a feature(unless that's required in someway)

File: src/Nancy.Templates/Nancy.Templates/Nancy.Templates.csproj [hunk 2]
```diff
@@ -15,7 +15,7 @@
     <AssemblyOriginatorKeyFile>Key.snk</AssemblyOriginatorKeyFile>
     <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
     <GeneratePkgDefFile>false</GeneratePkgDefFile>
-    <MinimumVisualStudioVersion>12.0</MinimumVisualStudioVersion>
+    <MinimumVisualStudioVersion>14.0</MinimumVisualStudioVersion>
```
Comment (devlead) [inline src/Nancy.Templates/Nancy.Templates/Nancy.Templates.csproj]: Correct me if I'm wrong but wouldn't this break the template on previous Visual Studio versions?

**Comment by devlead** [comment]
@jimmason I would fix the basics like rebasing and keeping the commit as clean as possible. Can't speak on behalf of maintainers, this repo doesn't have  a `contributing.md` but I bet it would follow the Nancy standard on avail [here](https://github.com/NancyFx/Nancy/blob/master/CONTRIBUTING.md)

I think with a few quick minor changes it would make it allot easier/less work for @thecodejunkie,  @grumpydev or one of the other maintainers to review and merge,

File: src/Nancy.Templates/Nancy.Templates/Nancy.Templates.csproj (hunk 2 above)
Comment (jimmason) [inline src/Nancy.Templates/Nancy.Templates/Nancy.Templates.csproj]: Yes, this shouldn't have gone in. Just got into a bit of a trouble on the git front when forgetting to check out a new branch and then pushing commits. 

My original intention was to submit a pull request containing only the change in 9b3f1d7. So i will try and get it back to this point.

Thanks for the feedback and apologies for making this one more complex than it should be.

**Comment by jimmason** [comment]
I have update the pull request to only include the changes which should have been present originally. Its still coming in from my master branch as i am not sure of a way to do this without this pull request being declined and a new request open.

Iv'e taken a good read through the guidelines and the contents have been duly noted.

Thanks again,

James

**Comment by danyx23** [comment]
Thanks for your work!

**Comment by devlead** [comment]
@jimmason excellent :+1: , now it's just up to the [`Custodians of the Super-Duper-Happy-Path`](https://github.com/orgs/NancyFx/people)  ;)

**Comment by kjpluck** [comment]
Coming in from #21:
I'm trying to convince new company to use nancy but tripping right on the very first hurdle. Please update the docs to not use templates until the templates are available.

**Comment by khellang** [comment]
> I'm trying to convince new company to use nancy but tripping right on the very first hurdle.

If you can't convince a company to use Nancy without project templates, I think you have bigger problems. The Nancy templates are basically just a simple blank "web application" with the Nancy package added through NuGet. It basically takes max. 2 mins to get to the starter template stage without them.

We're still looking at how we can handle these templates. It's a super manual process and it takes loads of time to maintain. In the meantime, you can look at http://sidewaffle.com/, they have Nancy templates as well. If you'd like to do some work, we'd appreciate it :smile:

> Please update the docs to not use templates until the templates are available.

Hey, the wiki is free to edit for anyone. If you see anything that isn't clear or that you think should be changed, go ahead! :sparkles:

**Comment by thecodejunkie** [comment]
Keep discussion in #21

**Comment by thecodejunkie** [comment]
To clarify why this was closed - neither of the open pull-requests are going to be pull in right now. I've written a longer answer https://github.com/NancyFx/Nancy.Templates/pull/21#issuecomment-157003366 as to why this is and #21 just happened to have a discussion with @devlead about automation

TL;DR just adding VS2015 support is not enough for us to maintain them, we need to find a way to automate the process so that the Nancy version can be kept up to date as well

Thanks for the PR, but this one falls on me =/

**Comment by jimmason** [comment]
No issues with that decsion on my part. It was purley something i did for my own usage and thought i would open an pr just in case it was useful for others. 
If there is anything i could help out with in the future feel free to reach out.
